modules/cert-manager-acme-dns01-ambient-gcp.adoc

```diff
@@ -0,0 +1,127 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-operator-issuer-acme.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-acme-dns01-ambient-gcp_{context}"]
+= Configuring an ACME issuer to solve DNS-01 challenges by using ambient credentials on GCP
+
+You can use the {cert-manager-operator} to set up an ACME issuer to solve DNS-01 challenges by using ambient credentials on GCP. This procedure uses _Let's Encrypt_ as the ACME CA server and shows how to solve DNS-01 challenges with Google CloudDNS.
+
+.Prerequisites
+
+* If you are using GCP with workload identity, you followed the instructions from the _Configuring cloud credentials for the cert-manager Operator for Red Hat OpenShift with GCP Workload Identity_ section.
+* If you are using GCP without workload identity, you followed the instructions from the _Configuring cloud credentials for the cert-manager Operator for Red Hat OpenShift on GCP_ section.
+
+.Procedure
+
+. Optional: Override the nameserver settings for the DNS-01 self check:
++
+This step is required only when the target public-hosted zone overlaps with the cluster's default private-hosted zone.
+
+.. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+.. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig:                                <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers-only'       <2>
+      - '--dns01-recursive-nameservers=1.1.1.1:53' <3>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
+<3> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS-01 self check.
+
+.. Save the file to apply the changes.
+
+. Optional: Create a namespace for the issuer:
++
+[source,terminal]
+----
+$ oc new-project my-issuer-namespace
+----
+
+. Modify the `CertManager` resource to add the `--issuer-ambient-credentials` argument:
++
+[source,terminal]
+----
+$ oc patch certmanager/cluster --type=merge -p='{"spec":{"controllerConfig":{"overrideArgs":["--issuer-ambient-credentials"]}}}'
+----
+
+. Create an issuer.
+
+.. Create a YAML file that defines the `Issuer` object:
++
+.Example `issuer.yaml` file
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: issuer-acme-dns01-clouddns <1>
+spec:
```

Comment:
Today happened to find one minor inconsistent issue, some `kind: Issuer` and `kind: Certificate` have `namespace: my-issuer-namespace` but some others have not. But this is a minor issue, not very big.

Reply:
Addressed